tree/treeplayer/inc/ROOT/TVecAllocator.hxx

```diff
@@ -0,0 +1,89 @@
+// Author: Enrico Guiraud, Enric Tejedor, Danilo Piparo CERN  01/2018
+
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_TVECALLOCATOR
+#define ROOT_TVECALLOCATOR
+
+#include <iostream>
+#include <memory>
+
+namespace ROOT {
+namespace Detail {
+namespace VecOps {
+
+template <typename T>
+class TVecAllocator {
```

Comment:
imho this allocator is of general interest and applicability, and is not bound to be used in conjunction with `TVec`, so I would give it a name that reflects what it does, like `TAdoptAllocator` or `TViewAllocator`

Reply:
Good point.
At this point we could move it out from Detail::VecOps and put it somewhere else... Not so sure this place is core/cont.

Reply:
Almost addressed. The new name TAdoptAllocator has been selected, the code documented, basically we are ready to move it to ROOT::Experimental if we want.